### PR TITLE
feat: add contour support to image viewer

### PR DESCRIPTION
**Pull request by astrofrog**
This allows users to then show either the bitmap, contour or both in the image viewer layers.

### CODE REVIEW

**Comment by maartenbreddels** [comment]
@astrofrog did you forgot to add/commit the glue_jupyter.bqplot.image.state module?

**Comment by astrofrog** [comment]
@maartenbreddels done!

File: glue_jupyter/bqplot/image/state.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class BqplotImageViewerState(ImageViewerState):
+
+    # TODO: change contour_visible to False by default once it can be toggled in UI
+
+    bitmap_visible = CallbackProperty(True, 'whether to show the image as a bitmap')
+    contour_visible = CallbackProperty(True, 'whether to show the image as contours')
+
+    mode = CallbackProperty('image', 'Whether to show an image or contour')
```
Comment (maartenbreddels) [inline glue_jupyter/bqplot/image/state.py:13]: I think it makes more sense to have this in the Layer state, and having both mode and the two visibilities sounds a bit duplicate right?

Comment (astrofrog) [reply glue_jupyter/bqplot/image/state.py:13]: Oops yes I intended to have this in the layer state, and indeed ``mode`` is left over from a previous experiment

**Comment by javerbukh** [comment]
@astrofrog Is this in a state where it can be tested with `jdaviz`? If so, what is the preferred way to add a `contour` to one of the viewers using this PR?

**Comment by astrofrog** [comment]
@javerbukh this is not yet ready/functional

**Comment by maartenbreddels** [comment]
Indeed, I can probably pick this up tomorrow.

**Comment by maartenbreddels** [comment]
Apart from some small issues (CI, and tests), this should be good for review, to test:

```
import glue_jupyter as gj
from glue.core import Data
import numpy as np
x, y = np.ogrid[-np.pi:np.pi:256j, -np.pi:np.pi:256j]
intensity = np.sin(np.exp((np.sin(x)**3 + np.cos(y)**2)))
data = Data()
data.add_component(intensity, label='intensity')

x, y = np.ogrid[-1:np.pi:256j, -1:np.pi:256j]
intensity2 = np.sin(np.exp((np.sin(x)**2 + np.cos(y)**3)))
data2 = Data()
data2.add_component(intensity2, label='intensity')

app = gj.jglue(data=data, data2=data2)
from glue.core.component_link import ComponentLink
link1 =  ComponentLink([data.id['Pixel Axis 0 [y]']], data2.id['Pixel Axis 0 [y]'])
link2 =  ComponentLink([data.id['Pixel Axis 1 [x]']], data2.id['Pixel Axis 1 [x]'])
app.data_collection.add_link([link1, link2])

v = app.imshow(data=data)
v.add_data(data2)
```

To see the contour, click the eye icon in the layer options next to the Contour text.

**Comment by javerbukh** [comment]
Tried contours with cubeviz and it looks great! As some of our product owners mentioned, it would be good to be able to have a contour map on image viewer 1 of the data in image viewer 2, I'm not sure if that's already possible but if not that would probably be the next contour feature they would request. Really great work guys!!

Example of it working on cubeviz:
Screen Shot 2020-09-10 at 9 42 42 AM

**Comment by maartenbreddels** [comment]
Yes, that should work. Basically in the demo code above, if you add this:
```python
l0, l1 = v.layers
l1.state.bitmap_visible = False
l1.state.contour_visible = True
```
You get:
image

**Comment by javerbukh** [comment]
Ah ok. So I tried adding the `IVAR` data to the same image viewer as `FLUX`, setting the `IVAR` layer to have contour on and its bitmap off, but I was not able to see the layer 2 contour map on top of layer 1. I tried using the code segment but still did not see contour from layer 2 and top of layer 1. The code segment does work for the sample code you sent, and I was able to get it working from the UI, so this could be outside the scope of glue-jupyter.

```
l0, l1 = viz.app.get_viewer("flux-viewer").layers
l1.state.bitmap_visible = False
l1.state.contour_visible = True
```

**Comment by javerbukh** [comment]
I just tested adding a 2d image to one of the image viewers, turning contour on and bitmap off, and it worked! I believe that was more important to have working than having the `ivar` contours overlaid on the `flux` image.

**Comment by maartenbreddels** [comment]
Great, we're doing some squashing/rebasing and this will be merged then!

**Comment by maartenbreddels** [comment]
This should be a replacement for https://github.com/spacetelescope/jdaviz/pull/199

Some CI fluke leftover (Linux job cancelled).

**Pull request merged**